sdk/core/core/inc/az_xml_parser.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_XML_PARSER_H
+#define AZ_XML_PARSER_H
+
+#include <az_xml_value.h>
+#include <az_result.h>
+#include <az_span_reader.h>
+#include <az_str.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct {
+  az_span name;
+  az_xml_value value;
+} az_xml_member;
+
+enum { AZ_XML_STACK_SIZE = 63 };
+
+typedef uint64_t az_xml_stack;
+
+typedef enum {
+  AZ_XML_STACK_OBJECT = 0,
+  AZ_XML_STACK_ARRAY = 1,
+} az_xml_stack_item;
+
+typedef struct {
+  az_span_reader reader;
+  az_xml_stack stack;
+} az_xml_parser;
+
+AZ_NODISCARD az_xml_parser az_xml_parser_create(az_span const buffer);
+
+AZ_NODISCARD az_result
+az_xml_parser_read(az_xml_parser * const self, az_xml_value * const out_value);
+
+AZ_NODISCARD az_result
+az_xml_parser_read_object_member(az_xml_parser * const self, az_xml_member * const out_member);
```

Comment:
I think, XML should have different interface. Like `read_attribute`, `read_item` etc.

Reply:
Yes, I was trying to implement a new design after the talk. Sorry I was occupied by other things. I'll be focusing on this from this week on.